TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/AuthScreen/AuthScreenViewController.m

```diff
@@ -0,0 +1,339 @@
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authFailed:) name:@"authfailed" object:nil];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authdone" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authCompleted:) name:@"authdone" object:nil];
+
+}
+
+
+-(void)viewWillDisappear:(BOOL)animated{
+    
+    [super viewWillDisappear:animated];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registerfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registrationdone" object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authdone" object:nil];
+
+}
+
+
+-(void)viewDidAppear:(BOOL)animated{
+    
+    [super viewDidAppear:animated];
+    
+}
+
+
+#pragma mark - app flow actions
+
+- (IBAction)btnAuthTapped:(id)sender{
+
+    NSLog(@"btnAuth");
+
+    if ([self checkProvidedValues] == true){
+        
+        [self showActivity];
+        
+        TASingleton *singleton = [[TASingleton alloc] init];
+        
+        if ([singleton isConnected] == true){
+            
+            NSString *userName = tfLogin.text;
+            NSString *password = tfPassword.text;
+            
+            BOOL modeLoginRegister = false;
+            modeLoginRegister = [swMode isOn];
+
+            if (modeLoginRegister == false){
+
+                [singleton authUser:userName password:password];
+
+            }else{
+
+                [singleton registerUser:userName password:password];
+
+            }
+            
+        }else{
+            
+            [self hideActivity];
+            
+        };
+        
+    }
+    
+}
+
+- (IBAction)swModeChanged:(id)sender{
+    
+    NSLog(@"swMode");
+
+    BOOL modeLoginRegister = false;
+    modeLoginRegister = [swMode isOn];
+    
+    if (modeLoginRegister == false){
+        //do login
+        
+        [btnAuth setTitle:@"LOG IN" forState:UIControlStateNormal];
+        
+    }else{
+        //do register
+
+        [btnAuth setTitle:@"REGISTER" forState:UIControlStateNormal];
+
+    }
+    
+}
+
+
+-(BOOL)checkLogin{
+    
+    BOOL result = false;
+    
+    if ([tfLogin.text length] > 0){
+        result = true;
+    }
+    
+    return result;
+}
+
+
+-(BOOL)checkPassword{
+    
+    BOOL result = false;
+    
+    if ([tfPassword.text length] > 0){
+        result = true;
+    }
+    
+    return result;
+}
+
+
+-(BOOL)checkProvidedValues{
+    
+    BOOL result = false;
+    int resultStatus = 0;
+    NSString *statusText = @"";
+    
+    if ([self checkLogin] == false){
+        resultStatus = resultStatus + 1;
+    }
+    
+    
+    if ([self checkPassword] == false){
+        resultStatus = resultStatus + 2;
+    }
+    
+    
+    switch (resultStatus) {
```

Comment:
Overly complicated. Given there's only one status shown it makes sense to `validate` the values one by one, update the status and return.

In a proper application I'd expect the statues to accumulate.